plugins/modules/ios_vrf_global.py

```diff
@@ -0,0 +1,207 @@
+        type: dict
+        suboptions:
+          multicast:
+            description: IP Multicast configuration
+            type: dict
+            suboptions:
+              multitopology:
+                description:  Enable Multicast-Specific RPF Topology
+                type: bool
+      rd:
+        description: Specify Route Distinguisher (RD).
+        type: str
+      route_target:
+        description: Specify Target VPN Extended Communities.
+        type: dict
+        suboptions:
+          export:
+            description: Export Target-VPN community.
+            type: str
+          import_config:
+            description: Export Target-VPN community.
+            type: str
+          both:
+            description: Both export and import Target-VPN community
+            type: str
+      vnet:
+        description: Virtual NETworking configuration.
+        type: dict
+        suboptions:
+          tag:
+            description: Identifier used to tag packets associated with this VNET
+            type: int
+      vpn:
+        description: Configure VPN ID for the VRF as specified in RFC 2685
+        type: dict
+        suboptions:
+          id:
+            description: Configure VPN ID in RFC 2685 format
+            type: str
+  running_config:
+    description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the IOS device by
+        executing the command B(show running-config vrf).
+      - The state I(parsed) reads the configuration from C(running_config) option and
+        transforms it into Ansible structured data as per the resource module's argspec
+        and the value is then returned in the I(parsed) key within the result.
+    type: str
+  state:
+    choices: [parsed, gathered, deleted, merged, replaced, rendered, overridden, purged]
+    default: merged
+    description:
+      - The state the configuration should be left in
+      - The states I(rendered), I(gathered) and I(parsed) does not perform any change
+        on the device.
+      - The state I(rendered) will transform the configuration in C(config) option to
+        platform specific CLI commands which will be returned in the I(rendered) key
+        within the result. For state I(rendered) active connection to remote host is
+        not required.
+      - The state I(gathered) will fetch the running configuration from device and transform
+        it into structured data in the format as per the resource module argspec and
+        the value is returned in the I(gathered) key within the result.
+      - The state I(parsed) reads the configuration from C(running_config) option and
+        transforms it into JSON format as per the resource module parameters and the
+        value is returned in the I(parsed) key within the result. The value of C(running_config)
+        option should be the same format as the output of command I(show running-config vrf).
```

Comment:
`show running-config vrf` is this correct for IOS-XE?

Reply:
@Ruchip16  use `sh running-config | section vrf` instead

Reply:
@rohitthakur2590 with all the Cisco platforms we generally follow using `| section <resource>`
In many cases `show running-config vrf` gave incomplete information or we had issues with how simple configuration is rendered as part of config and showed with this pattern of config fetching.
Using the | we can ensure the running configuration is stripped out guaranteeing the configuration we fetch is aligned with the configuration we would be pushing. For reference, we had to do a lot of refactoring for the ios_acls module as the module used to `show access-list` command instead of the pure running-config data.

Reply:
Also, brings in a lot of unnecessary lines which we have to deal with in our facts churning step.
```
Router3#show running-config | section vrf
vrf definition ansible
vrf definition test
 vnet tag 22
 description test vrf ansible
 ipv4 multicast multitopology
 rd 10.2.3.4:300
 vpn id 3:4
Router3#show running-config vrf 
Building configuration...

Current configuration : 154 bytes
vrf definition ansible
!
vrf definition test
 vnet tag 22
 description test vrf ansible
 ipv4 multicast multitopology
 rd 10.2.3.4:300
 vpn id 3:4
!
end
```